overlord/snapstate/snapmgr.go

```diff
@@ -196,13 +199,17 @@ type ComponentSetup struct {
 	CompType snap.ComponentType
 	// CompPath is the path to the file
 	CompPath string `json:"comp-path,omitempty"`
+	// UnlinkedComp has information about the unlinked component in
+	// component refreshes
+	UnlinkedComp *sequence.ComponentState `json:"unlinked-comp,omitempty"`
```

Comment:
don't we actually just need to ComponentSideInfo part of that? but also same consideration

Reply:
We need the full ComponentState as this is used when undoing the unlink too